In [5]:
import pandas as pd

In [6]:
# API Docs
import dotenv, os
from dune_client.types import QueryParameter
from dune_client.client import DuneClient
from dune_client.query import QueryBase

# POST /v1/query/{query_id}/execute

# GET /v1/execution/{execution_id}/results

In [7]:
# change the current working directory where .env file lives
os.chdir("/Users/Stephanie/Desktop/Folder/Analytics/eth_etf/eth_etf")
# load .env file
dotenv.load_dotenv("api_key.env")
# setup Dune Python client
dune = DuneClient.from_env()

SWAPS BY DATE

In [8]:
requested_date = "2024-05-20"
query = QueryBase(
    query_id=3767801,
      params=[
        QueryParameter.text_type(name="date_parameter", value=requested_date),
    ],
)

query_result = dune.run_query_dataframe(
  query=query
) 

2024-05-28 21:45:09,806 INFO dune_client.api.base executing 3767801 on medium cluster
2024-05-28 21:45:10,505 INFO dune_client.api.base waiting for query execution 01HZ13SD4W4F8NGXDT54P24F4Q to complete: ExecutionState.PENDING (queue position: 2)
2024-05-28 21:45:11,648 INFO dune_client.api.base waiting for query execution 01HZ13SD4W4F8NGXDT54P24F4Q to complete: ExecutionState.EXECUTING


In [9]:
may_20_df = pd.DataFrame(query_result)
may_20_df['Total USD'] = pd.to_numeric(may_20_df['Total USD'], errors='coerce').fillna(0).astype(float).astype('int64')
may_20_df

token_bought_symbol   Total USD
0                   WETH  4853837606
1                   USDC  1912623889
2                   USDT  1426194196
3                   WBTC   539018387
4                   USDe   283575301
...                  ...         ...
1065             μCloneX           0
1066               michi           0
1067             $HAMSTR           0
1068                μWOW           0
1069                 KOK           0

[1070 rows x 2 columns]

In [10]:
requested_date = "2024-05-19"
query = QueryBase(
    query_id=3767801,
      params=[
        QueryParameter.text_type(name="date_parameter", value=requested_date),
    ],
)

query_result = dune.run_query_dataframe(
  query=query
)

2024-05-28 21:45:13,089 INFO dune_client.api.base executing 3767801 on medium cluster
2024-05-28 21:45:13,353 INFO dune_client.api.base waiting for query execution 01HZ13SFYE2Z839D646172GW3B to complete: ExecutionState.PENDING (queue position: 2)


In [11]:
may_19_df = pd.DataFrame(query_result)
may_19_df['Total USD'] = pd.to_numeric(may_19_df['Total USD'], errors='coerce').fillna(0).astype(float).astype('int64')
may_19_df

token_bought_symbol   Total USD
0                  WETH  1608726665
1                  USDC   783929095
2                  USDT   777504831
3                wstETH   206971489
4                  WBTC   165243880
..                  ...         ...
993             μCloneX           0
994                 WIF           0
995              gtcETH           0
996               μELEM           0
997                  μ⚇           0

[998 rows x 2 columns]

In [13]:
may_20_total = may_20_df['Total USD'].sum()
may_20_df['% TTL'] = (may_20_df['Total USD'] / may_20_total) * 100
may_20_df

token_bought_symbol   Total USD      % TTL
0                   WETH  4853837606  47.109253
1                   USDC  1912623889  18.563102
2                   USDT  1426194196  13.842025
3                   WBTC   539018387   5.231480
4                   USDe   283575301   2.752259
...                  ...         ...        ...
1065             μCloneX           0   0.000000
1066               michi           0   0.000000
1067             $HAMSTR           0   0.000000
1068                μWOW           0   0.000000
1069                 KOK           0   0.000000

[1070 rows x 3 columns]

In [14]:
may_19_total = may_19_df['Total USD'].sum()
may_19_df['% TTL'] = (may_19_df['Total USD'] / may_19_total) * 100
may_19_df

token_bought_symbol   Total USD      % TTL
0                  WETH  1608726665  37.553886
1                  USDC   783929095  18.299929
2                  USDT   777504831  18.149962
3                wstETH   206971489   4.831513
4                  WBTC   165243880   3.857430
..                  ...         ...        ...
993             μCloneX           0   0.000000
994                 WIF           0   0.000000
995              gtcETH           0   0.000000
996               μELEM           0   0.000000
997                  μ⚇           0   0.000000

[998 rows x 3 columns]

In [16]:
may_20_weth = may_20_df[may_20_df['token_bought_symbol'] == 'WETH']
may_20_weth

token_bought_symbol   Total USD      % TTL
0                WETH  4853837606  47.109253

In [17]:
may_19_weth = may_19_df[may_19_df['token_bought_symbol'] == 'WETH']
may_19_weth

token_bought_symbol   Total USD      % TTL
0                WETH  1608726665  37.553886

In [23]:
increase = may_20_weth['Total USD'] - may_19_weth['Total USD']
increase_value = increase.iloc[0]
format_increase = "{:,}".format(increase_value)
format_increase

'3,245,110,941'

On May 20th, news came out suggesting that the SEC was now likely to approve the Ethereum ETF. On this day, Ethereum spiked 20% and closed at $3,691.

This analysis looks into swaps into ETH on the Ethereum chain on Uniswap.

May 20th, $4.8 billion dollars swapped into Ethereum vs. $1.6 billion the day before, an increase of $3.2 billion or 200%. In addition, the % of Total swaps going into ETH increased 10 points, from 37% to 47%.